trl/trainer/ppo_trainer.py

```diff
@@ -1297,6 +1297,16 @@ def log_stats(
             if not isinstance(rewards, torch.Tensor):
                 rewards = torch.tensor(rewards).to(self.current_device)
 
+            # Gather rewards if distributed
+            if self.is_distributed:
+                import torch.distributed as dist
+
+                dist.barrier()
+                world_size = dist.get_world_size()
+                tensors = [torch.zeros_like(rewards) for _ in range(world_size)]
+                dist.all_gather(tensors, rewards)
+                rewards = torch.cat(tensors)
```

Comment:
Maybe all these line of code can be replaced with self.accelerator.gather(rewards)


```suggestion
            # Gather rewards if distributed
            rewards = self.accelerator.gather(rewards).flatten()
```

Reply:
Ah nice, I was not aware accelerate has this helper method, I will clean up.